dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchCollection.cs

```diff
@@ -81,13 +83,9 @@ public AzureAISearchCollection(SearchIndexClient searchIndexClient, string name,
             new AzureAISearchDynamicModelBuilder().Build(typeof(TRecord), options.VectorStoreRecordDefinition, options.EmbeddingGenerator) :
             new AzureAISearchModelBuilder().Build(typeof(TRecord), options.VectorStoreRecordDefinition, options.EmbeddingGenerator, options.JsonSerializerOptions);
 
-        // Resolve mapper.
-        // If they didn't provide a custom mapper, and the record type is the generic data model, use the built in mapper for that.
-        // Otherwise, don't set the mapper, and we'll default to just using Azure AI Search's built in json serialization and deserialization.
-        if (typeof(TRecord) == typeof(Dictionary<string, object?>))
-        {
-            this._dynamicMapper = new AzureAISearchDynamicMapper(this._model);
-        }
+        this._mappper = typeof(TRecord) == typeof(Dictionary<string, object?>) ?
+            (new AzureAISearchDynamicMapper(this._model, options.JsonSerializerOptions) as IAzureAISearchMapper<TRecord>)! :
```

Comment:
nit: alternative syntax for this that's IMHO slightly better (but not by much):

```suggestion
            (IAzureAISearchMapper<TRecord>)(object)new AzureAISearchDynamicMapper(this._model, options.JsonSerializerOptions) :
```

(this doesn't require a bang (!) and if the cast fails, this immediately throws InvalidCastOperation instead of NullReferenceException later (though this should never happen))

Reply:
I really don't think it matters, but sure.